projects/pastanaga-angular/src/lib/side-nav/side-nav.component.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { CommonModule } from '@angular/common';
+import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
+import { MockModule } from 'ng-mocks';
+import { SideNavComponent } from './side-nav.component';
+import { PaButtonModule } from '../button/button.module';
+import { ReplaySubject, Subject } from 'rxjs';
+import { TranslatePipe } from '../translate/translate.pipe';
+import { PastanagaService } from '../pastanaga.service';
+import { createSpyObject, mockProvider } from '@ngneat/spectator';
+import { BreakpointObserver, ViewportMode } from '../breakpoint-observer/breakpoint.observer';
+import { fakeAsync, tick } from '@angular/core/testing';
+
+describe('SideNavComponent', () => {
+    let spectator: Spectator<SideNavComponent>;
+    let component: SideNavComponent;
+    const currentMode: ReplaySubject<ViewportMode> = new ReplaySubject<ViewportMode>(1);
+    const toggle: Subject<boolean> = new Subject<boolean>();
+    const createComponent = createComponentFactory({
+        component: SideNavComponent,
+        imports: [MockModule(CommonModule), MockModule(PaButtonModule)],
+        mocks: [TranslatePipe],
+        providers: [
+            mockProvider(PastanagaService, {
+                breakpoint: createSpyObject(BreakpointObserver, {
+                    currentMode,
+                }),
+            }),
+        ],
+        detectChanges: false,
+    });
+
+    beforeEach(() => {
+        spectator = createComponent();
+        component = spectator.component;
+        component.toggle = toggle;
+    });
+
+    describe('when visible is true', () => {
+        describe(`desktop`, () => {
+            beforeEach(() => {
+                currentMode.next('desktop');
+                spectator.detectComponentChanges();
+            });
+            it(`should display a navbar`, () => {
+                expect(spectator.query('.pa-side-nav')).toBeTruthy();
+            });
+            it(`should not have an overlay`, () => {
+                expect(spectator.query('.pa-side-nav-tablet-overlay')).toBe(null);
+            });
+            it(`should not display close button`, () => {
+                expect(spectator.query('.pa-close-side-nav-button')).toBe(null);
+            });
+        });
+
+        describe(`tablet`, () => {
+            beforeEach(() => {
+                currentMode.next('tablet');
+                component.visible = true;
```

Comment:
your first describe block is `when visible is true`, so this `component.visible = true;` should go in a beforeEach block of this first describe instead of having it in both tablet and mobile.

Reply:
i need to set it to true after the current mode since if it's not desktop visible will be false